tests/cocotb/test_tcdm_subsys.py

```diff
@@ -0,0 +1,181 @@
+# ---------------------------------
+# Copyright 2023 KULeuven
+# Solderpad Hardware License, Version 0.51, see LICENSE for details.
+# SPDX-License-Identifier: SHL-0.51
+# Author: Ryan Antonio (ryan.antonio@esat.kuleuven.be)
+#
+# Description:
+# This tests the basic read and write funtcionality for each
+# input port and checks if we can read/write data into the TCDM
+# ---------------------------------
+
+import cocotb
+from cocotb.triggers import RisingEdge, Timer
+from cocotb.clock import Clock
+from cocotb_test.simulator import run
+from decimal import Decimal
+import pytest
+import snax_util
+
+# Configurable design time parameters
+NARROW_DATA_WIDTH = 64
+TCDM_DEPTH = 64
+NR_BANKS = 8
+NUM_INPUT = 2
+
+# Optinally configurable parameters
+NUM_OUTPUT = NR_BANKS
+MIN_VAL = 0
+MAX_VAL = 2**NARROW_DATA_WIDTH
+BANK_INCREMENT = NARROW_DATA_WIDTH / 8
```

Comment:
The defaults were just so that the test is smaller 😄 but it should be fine if we go larger.

I put it here so that even the stimuli after the line in `async def tcdm_subsys_dut(dut):` also use the same parameters. And not all parameters go into the parameters of the module.